contracts/modules/cover/Cover.sol

```diff
@@ -290,8 +301,15 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
       )
     );
 
-    // TODO: implement using buckets
-    totalActiveCoverInAsset[params.coverAsset] += coverAmountInCoverAsset;
+    // Update totalActiveCover
+    uint currentBucketId = block.timestamp / BUCKET_SIZE;
+    // TODO: delete expired buckets?
+    expiredCoverAmount += getExpiredCoverAmount(params.coverAsset, lastBucketUpdateId, currentBucketId);
```

Comment:
When `lastBucketUpdateId == currentBucketId`, the `getExpiredCoverAmount` function will return the amount for the current bucket. This seems to subtract the current bucket at every cover buy. Needs a check here to make sure you do only once. Ideally I would inline the while loop here and avoid generating a public function because (1) we already have one that is autogenerated and (2) contract size got to 24k, terribly close to the limit.